src/main/java/org/folio/circulation/infrastructure/storage/inventory/LocationRepository.java

```diff
@@ -60,15 +60,23 @@ public static LocationRepository using(Clients clients) {
       return ofAsync(() -> null);
     }
 
-    return SingleRecordFetcher.json(locationsStorageClient, "location",
-      response -> succeeded(null))
-      .fetch(item.getLocationId())
-      .thenApply(r -> r.map(Location::from))
+    return fetchLocationById(item.getLocationId())
       .thenCompose(r -> r.after(this::loadLibrary))
       .thenCompose(r -> r.after(this::loadCampus))
       .thenCompose(r -> r.after(this::loadInstitution));
   }
 
+  public CompletableFuture<Result<Location>> fetchLocationById(String id) {
+    if(StringUtils.isBlank(id)) {
+      return ofAsync(() -> null);
+    }
```

Comment:
Is it possible to write a unit test to check this guard clause?

Reply:
Sure. added.